hummingbot/strategy/avellaneda_market_making/avellaneda_market_making_config_map_pydantic.py

```diff
@@ -0,0 +1,529 @@
+        cls, v: Union[str, InfiniteModel, FromDateToDateModel, DailyBetweenTimesModel]
+    ):
+        if not isinstance(v, Dict) and not hasattr(ExecutionTimeframe, v):
+            raise ValueError(
+                f"Invalid timeframe, please choose value from {[e.value for e in list(ExecutionTimeframe)]}"
+            )
+        elif isinstance(v, (InfiniteModel, FromDateToDateModel, DailyBetweenTimesModel)):
+            sub_model = v
+        elif v == ExecutionTimeframe.infinite:
+            sub_model = InfiniteModel.construct()
+        elif v == ExecutionTimeframe.daily_between_times:
+            sub_model = DailyBetweenTimesModel.construct()
+        elif v == ExecutionTimeframe.from_date_to_date:
+            sub_model = FromDateToDateModel.construct()
+        else:  # isinstance(v, Dict)
+            sub_model = v
+        return sub_model
+
+    @validator("order_refresh_tolerance_pct", pre=True)
+    def validate_order_refresh_tolerance_pct(cls, v: str):
+        ret = validate_decimal(v, min_value=Decimal("-10"), max_value=Decimal("10"), inclusive=True)
+        if ret is not None:
+            raise ValueError(ret)
+        return v
+
+    @validator("volatility_buffer_size", "trading_intensity_buffer_size", pre=True)
+    def validate_buffer_size(cls, v: str):
+        ret = validate_int(v, 1, 10_000)
+        if ret is not None:
+            raise ValueError(ret)
+        return v
+
+    @validator("order_levels_mode", pre=True)
+    def validate_order_levels_mode(cls, v: Union[str, SingleOrderLevelModel, MultiOrderLevelModel]):
+        if not isinstance(v, Dict) and not hasattr(OrderLevelsMode, v):
+            raise ValueError(
+                f"Invalid order levels mode, please choose value from {[e.value for e in list(OrderLevelsMode)]}."
+            )
+        elif isinstance(v, (SingleOrderLevelModel, MultiOrderLevelModel)):
+            sub_model = v
+        elif v == OrderLevelsMode.single_order_level:
+            sub_model = SingleOrderLevelModel.construct()
+        elif v == OrderLevelsMode.multi_order_level:
+            sub_model = MultiOrderLevelModel.construct()
+        else:  # isinstance(v, Dict)
+            sub_model = v
+        return sub_model
+
+    @validator("hanging_orders_mode", pre=True)
+    def validate_hanging_orders_mode(cls, v: Union[str, TrackHangingOrdersModel, IgnoreHangingOrdersModel]):
+        if not isinstance(v, Dict) and not hasattr(HangingOrdersMode, v):
+            raise ValueError(
+                f"Invalid hanging order mode, please choose value from {[e.value for e in list(HangingOrdersMode)]}."
+            )
+        elif isinstance(v, (TrackHangingOrdersModel, IgnoreHangingOrdersModel)):
+            sub_model = v
+        elif v == HangingOrdersMode.track_hanging_orders:
+            sub_model = TrackHangingOrdersModel.construct()
+        elif v == HangingOrdersMode.ignore_hanging_orders:
+            sub_model = IgnoreHangingOrdersModel.construct()
```

Comment:
Verify also here the comments made for ExecutionTimeFrame validator

Reply:
Addressed